src/client/mod.rs

```diff
@@ -288,23 +309,44 @@ pub struct WrapData {
+    pub local_secret_ids: bool,
     /// Comma-separated list of CIDR blocks; if set, specifies blocks of IP addresses which can
     /// perform the login operation.
-    pub bound_cidr_list: String,
-    /// If set, the token generated using this `AppRole` is a periodic token; so long as it is
-    /// renewed it never expires, but the TTL set on the token at each renewal is fixed to the value
-    /// specified here. If this value is modified, the token will pick up the new value at its next
-    /// renewal.
-    pub period: VaultDuration,
-    /// List of policies set on tokens issued via this `AppRole`.
-    pub policies: Vec<String>,
+    pub secret_id_bound_cidrs: Option<Vec<String>>,
     /// Number of times any particular `SecretID` can be used to fetch a token from this `AppRole`,
     /// after which the `SecretID` will expire.
     pub secret_id_num_uses: u64,
-    /// Duration after which any `SecretID` expires.
+    /// Duration in either an integer number of seconds (3600) or an integer time unit (60m) after which any SecretID expires.
     pub secret_id_ttl: VaultDuration,
+    /// List of CIDR blocks; if set, specifies blocks of IP addresses which can authenticate successfully,
+    /// and ties the resulting token to these blocks as well.
+    pub token_bound_cidrs: Vec<String>,
+    /// If set, will encode an explicit max TTL onto the token. This is a hard cap even if token_ttl and
+    /// token_max_ttl would otherwise allow a renewal.
+    pub token_explicit_max_ttl: VaultDuration,
+    /// If set, the default policy will not be set on generated tokens; otherwise it will be added to
+    /// the policies set in token_policies.
+    pub token_no_default_policy: bool,
     /// Duration after which the issued token can no longer be renewed.
     pub token_max_ttl: VaultDuration,
+    /// The maximum number of times a generated token may be used (within its lifetime); 0 means unlimited.
+    pub token_num_uses: u64,
+    /// The incremental lifetime for generated tokens.
+    /// If set, the token generated using this `AppRole` is a periodic token; so long as it is
+    /// renewed it never expires, but the TTL set on the token at each renewal is fixed to the value
+    /// specified here. If this value is modified, the token will pick up the new value at its next
+    /// renewal.
+    pub token_period: VaultDuration,
+    /// List of policies to encode onto generated tokens.
+    pub token_policies: Vec<String>,
+    /// The incremental lifetime for generated tokens.
+    pub token_ttl: VaultDuration,
+    /// The type of token that should be generated. Can be service, batch, or default to use the mount's
+    /// tuned default (which unless changed will be service tokens). For token store roles, there are two
+    /// additional possibilities: default-service and default-batch which specify the type to return unless
+    /// the client requests a different type at generation time.
+    pub token_type: String,
```

Comment:
This should be an enum rather than a String

Reply:
Cool, will do!